### DISCUSSION TITLE
Hero's skill aquire algorithm

### DISCUSSION THREAD

**Discussion by crypt47** [ihhub/fheroes2#8595] [General]
### Your question

My personal fun is to beaf up heroes with the skills I like. I often replay the game if lousy skill drops. I would like to know if the order of the skills is pregenerated on some point (seeding) or there is a way to influence the game engine to produce another skill? Thanks.

**Comment by oleg-derevenetz** [comment]
The order of skills depends on the initial map seed (which is generated when the new game is started) and pre-defined for a given hero. See this method for details and comments:

https://github.com/ihhub/fheroes2/blob/4fd43fb5356207581ca65e06440b0c262d8425b3/src/fheroes2/heroes/heroes.cpp#L2170-L2201

**Comment by crypt47** [comment]
Yes, I see. The only way to change upcomming choice is to upgrade hero skills without raising experience. It's a pitty that choice doesn't necessarily include one skill that was already selected before.

**Comment by ihhub** [comment]
Moving this to the discussion section.

**Reply by crypt47** [reply]
@oleg-derevenetz regarding my previous question in this topic. I think, maybe I'll manage to patch the game a bit for myself. Could you point me to the header file where secondary skill's IDs are defined? Which method can I use to check if the current hero is a player or a computer?

**Reply by oleg-derevenetz** [reply]
Please see the `src/fheroes2/heroes/skill.h` for skill IDs and `isControlAI()` method of `Heroes` class (ultimately inherited from the `Control` class) to check if a hero is controlled by AI or not.

**Reply by crypt47** [reply]
Thank you, I removed for a player MYSTICISM, EAGLE_EYE, SCOUTING and ESTATES as these skills needs to be beefed up to be of any interest. I also removed NECROMANCY as I don't play for undeads. I also tried to put a condition so that already acquired skills will grow up, but for some reason it didn't work. https://github.com/ihhub/fheroes2/blob/master/src/fheroes2/heroes/skill.cpp#L774

`
    	    if ( skill.Level() == Level::NONE && skill.Skill() == Skill::Secondary::MYSTICISM ) {
    		blacklist.insert( Skill::Secondary::MYSTICISM );
	    }
`
